app/api/v1/projects.rb

```diff
@@ -0,0 +1,65 @@
+class V1::Projects < Grape::API
+  resource :projects do
+    helpers do
+      def authorize_project!
+        return unless current_team
+        @allowed_projects = current_team.projects.pluck(:slug)
+
+        return unless params[:slug]
+        return if @allowed_projects.include?(params[:slug])
+
+        error!('Unauthorized. Invalid team.', 401)
+      end
+    end
+
+    before do
+      authenticate!
+      authorize_project!
+    end
+
+    desc 'Return all projects', { tags: ['project'] }
+    paginate
+    get '/' do
+      projects = Project.all
```

Comment:
I'm not familiar with Grape, but seems that would be a great idea to use Pundit on API endpoints too.